packages/back-end/src/models/ImpactEstimateModel.ts

```diff
@@ -66,11 +66,7 @@ export async function getImpactEstimate(
 
   let segmentObj: SegmentInterface | null = null;
   if (segment) {
-    segmentObj = await SegmentModel.findOne({
-      id: segment,
-      organization,
-      datasource: datasource.id,
-    });
+    segmentObj = await findSegmentById(segment, organization);
```

Comment:
Not sure if there is a use-case where we might also want to pass in the datasource.id like we were doing before. It feels redundant, but maybe there is a reason for it I'm not aware of.